pandas/core/interval.py

```diff
@@ -0,0 +1,413 @@
+
+    @cache_readonly
+    def is_unique(self):
+        return self._multiindex.is_unique
+
+    @cache_readonly
+    def is_non_overlapping_monotonic(self):
+        # must be increasing  (e.g., [0, 1), [1, 2), [2, 3), ... )
+        # or decreasing (e.g., [-1, 0), [-2, -1), [-3, -2), ...)
+        # we already require left <= right
+        return ((self.right[:-1] <= self.left[1:]).all() or
+                (self.left[:-1] >= self.right[1:]).all())
+
+    def _convert_scalar_indexer(self, key, kind=None):
+        return key
+
+    def _maybe_cast_slice_bound(self, label, side, kind):
+        return getattr(self, side)._maybe_cast_slice_bound(label, side, kind)
+
+    def _convert_list_indexer(self, keyarr, kind=None):
+        """
+        we are passed a list indexer.
+        Return our indexer or raise if all of the values are not included in the categories
+        """
+        locs = self.get_indexer(keyarr)
+        # TODO: handle keyarr if it includes intervals
+        if (locs == -1).any():
+            raise KeyError("a list-indexer must only include existing intervals")
+
+        return locs
+
+    def _check_method(self, method):
+        if method is not None:
+            raise NotImplementedError(
+                'method %r not yet implemented for IntervalIndex' % method)
+
+    def _searchsorted_monotonic(self, label, side, exclude_label=False):
+        if not self.is_non_overlapping_monotonic:
+            raise KeyError('can only get slices from an IntervalIndex if '
+                           'bounds are non-overlapping and all monotonic '
+                           'increasing or decreasing')
+
+        if isinstance(label, IntervalMixin):
+            raise NotImplementedError
+
+        if ((side == 'left' and self.left.is_monotonic_increasing) or
+                (side == 'right' and self.left.is_monotonic_decreasing)):
+            sub_idx = self.right
+            if self.open_right or exclude_label:
+                label = _get_next_label(label)
+        else:
+            sub_idx = self.left
+            if self.open_left or exclude_label:
+                label = _get_prev_label(label)
+
+        return sub_idx._searchsorted_monotonic(label, side)
+
+    def _get_loc_only_exact_matches(self, key):
+        return self._multiindex._tuple_index.get_loc(key)
+
+    def _find_non_overlapping_monotonic_bounds(self, key):
+        if isinstance(key, IntervalMixin):
+            start = self._searchsorted_monotonic(
+                key.left, 'left', exclude_label=key.open_left)
+            stop = self._searchsorted_monotonic(
+                key.right, 'right', exclude_label=key.open_right)
+        else:
+            # scalar
+            start = self._searchsorted_monotonic(key, 'left')
+            stop = self._searchsorted_monotonic(key, 'right')
+        return start, stop
+
+    def get_loc(self, key, method=None):
+        self._check_method(method)
+
+        original_key = key
```

Comment:
you shouldn't really be defining ``.get_loc``, it should just use ``.get_indexer`` which does all of the heavy lifting. is there a reason?